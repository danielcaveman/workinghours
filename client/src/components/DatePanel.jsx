import React from "react";
import styled from "styled-components";
import { useSelector, useDispatch } from "react-redux";
import { DatePicker } from "@material-ui/pickers";
import { Paper, Typography } from "@material-ui/core";
import { changeDate } from "../actions/DatePanel.actions";

const StyledContainer = styled(Paper)`
  margin: 1rem 0;
  padding: 2rem;
`;

function DatePanel() {
  const datePanel = useSelector((state) => state.datePanel);
  const dispatch = useDispatch();
  return (
    <StyledContainer>
      <Typography variant="h6" gutterBottom>
        Choose a Date:
      </Typography>
      <DatePicker
        views={["year", "month"]}
        label="Date"
        value={datePanel.date}
        onChange={(e) => dispatch(changeDate(e._d))}
        minDate={new Date("1990-02-01")}
        maxDate={new Date("2100-02-01")}
      />
    </StyledContainer>
  );
}

export default DatePanel;
