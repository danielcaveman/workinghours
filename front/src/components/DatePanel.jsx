import { useReducer } from "react";
import styled from "styled-components";
import { DateRangeInput } from "@datepicker-react/styled";

const Panel = styled.div`
  display: flex;
  background-color: #ffffff;
  border: 1px solid #dddddd;
  border-bottom: none;
  padding: 2rem;
  font-size: 1.6rem;
`;

const initialState = {
  startDate: null,
  endDate: null,
  focusedInput: null,
};

function reducer(state, action) {
  switch (action.type) {
    case "focusChange":
      return { ...state, focusedInput: action.payload };
    case "dateChange":
      return action.payload;
    default:
      throw new Error();
  }
}

function DatePanel() {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Panel>
      <DateRangeInput
        onDatesChange={(data) =>
          dispatch({ type: "dateChange", payload: data })
        }
        onFocusChange={(focusedInput) =>
          dispatch({ type: "focusChange", payload: focusedInput })
        }
        startDate={state.startDate} // Date or null
        endDate={state.endDate} // Date or null
        focusedInput={state.focusedInput} // START_DATE, END_DATE or null
      />
    </Panel>
  );
}

export default DatePanel;
