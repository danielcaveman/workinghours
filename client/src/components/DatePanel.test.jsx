import React from "react";
import { Provider } from "react-redux";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";
import MomentUtils from "@date-io/moment";
import configureStore from "redux-mock-store";
import { render, screen } from "@testing-library/react";
import DatePanel from "./DatePanel";

describe("<DatePanel />", () => {
  it("Should render DatePanel component correctly", () => {
    const initialState = { datePanel: { date: new Date() } };
    const mockStore = configureStore();
    const store = mockStore(initialState);
    render(
      <MuiPickersUtilsProvider utils={MomentUtils}>
        <Provider store={store}>
          <DatePanel />
        </Provider>
      </MuiPickersUtilsProvider>
    );

    const Text = screen.getByRole("heading", { name: /Choose a Date/i });
    expect(Text).toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const initialState = { datePanel: { date: new Date() } };
    const mockStore = configureStore();
    const store = mockStore(initialState);

    const { container } = render(
      <MuiPickersUtilsProvider utils={MomentUtils}>
        <Provider store={store}>
          <DatePanel />
        </Provider>
      </MuiPickersUtilsProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
