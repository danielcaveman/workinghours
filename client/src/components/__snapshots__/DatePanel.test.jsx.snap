// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePanel /> Should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root sc-bdfBwQ lnrRBG MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
    >
      Choose a Date:
    </h6>
    <div
      class="MuiFormControl-root MuiTextField-root"
      format="MMMM YYYY"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink="true"
      >
        Date
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          readonly=""
          type="text"
          value="March 2021"
        />
      </div>
      
    </div>
  </div>
</div>
`;
